### PR TITLE
MAYA-122014 support duplicate to Maya to non root

### PR DESCRIPTION
**Pull request by pierrebai-adsk**
- Add isMayaRootPath helper function.
- Use the destination path when duplicating as Maya.
- Add a unit test.

### CODE REVIEW

**Comment by pierrebai-adsk** [comment]
The only PF failure is the Linux interactive on GPU which is currently known to fail due to image comparison.

File: lib/mayaUsd/ufe/Utils.cpp [hunk 1]
```diff
@@ -372,6 +372,21 @@ MDagPath ufeToDagPath(const Ufe::Path& ufePath)
     );
 }
 
+bool isMayaRootPath(const Ufe::Path& ufePath)
```
Comment (ppt-adsk) [inline lib/mayaUsd/ufe/Utils.cpp:375]: Could be simply 

return (ufePath.runTimeId() == g_MayaRtid) && (ufePath.size() == 1);

File: lib/mayaUsd/ufe/Utils.h [hunk 2]
```diff
@@ -125,6 +125,10 @@ Ufe::PathSegment dagPathToPathSegment(const MDagPath& dagPath);
 MAYAUSD_CORE_PUBLIC
 MDagPath ufeToDagPath(const Ufe::Path& ufePath);
 
+//! Verify if the UFE path is the Maya root.
```
Comment (ppt-adsk) [inline lib/mayaUsd/ufe/Utils.h:128]: Hmmm, wonder about the name.  In Maya the root node is called world, and it's called that way in Maya commands and documentation as well.  Perhaps call it isMayaWorldPath()?

File: test/lib/mayaUsd/fileio/testDuplicateAs.py [hunk 3]
```diff
@@ -86,6 +86,39 @@ def testDuplicateAsMaya(self):
...
+        (_, _, aXlation, aUsdUfePathStr, aUsdUfePath, _, _, 
+               bXlation, bUsdUfePathStr, bUsdUfePath, _) = \
+            createSimpleXformScene()
+
+        xform = cmds.createNode('transform')
+        xformNames = cmds.ls(xform)
+        self.assertEqual(1, len(xformNames))
+        xformName = xformNames[0]
+
+        # Duplicate USD data as Maya data, placing it under the root.
```
Comment (ppt-adsk) [inline test/lib/mayaUsd/fileio/testDuplicateAs.py:101]: Comment is incorrect.

**Review by ppt-adsk — CHANGES_REQUESTED** [review]
Thanks for the fix!  Minor tweaks only.

**Review by ppt-adsk — APPROVED** [review]

**Comment by pierrebai-adsk** [comment]
Again, image diff on linux error. We can ignore, unrelated to this PR.

**Pull request merged**